app/background.ts

```diff
@@ -72,12 +72,15 @@ class BackgroundWorker {
       this.data[this.currentTab].enabled &&
       this.currentTab === details.tabId
     ) {
-      tabRecords.requests.push(details);
-      this.data[this.currentTab] = tabRecords;
-      chrome.browserAction.setBadgeText({
-        text: `${tabRecords.requests.length}`,
-        tabId: details.tabId
-      });
+        if (tabRecords.requests.filter((req) => req.requestId === details.requestId).length > 0) {
+          return
+        }
+        tabRecords.requests.push(details);
+        this.data[this.currentTab] = tabRecords;
+        chrome.browserAction.setBadgeText({
```

Comment:
Why we need to update the badge twice? beginning of this callback also we are setting  badge.